### ISSUE TITLE
Can't pass bool value to `diagnostics.signs` to disable diagnostics signs

### ISSUE THREAD

**Issue by Innei** [neovim/nvim-lspconfig#3113]
### Description

I set `diagnostics.signs` to false, but got an error. This came after a version update. Hope you reply, thank you

```lua
  opts.diagnostics = vim.tbl_extend("force", opts.diagnostics, {
        underline = true,
        update_in_insert = true,
        virtual_text = false,
        severity_sort = true,
        signs = false, // this line
      })

```

```
 | lazy.nvim:  | lazy.nvim: Failed to run `config` for nvim-lspconfig                                             
                                                                                                                    
...share/nvim/lazy/LazyVim/lua/lazyvim/plugins/lsp/init.lua:123: attempt to index field 'signs' (a boolean value)   
                                                                                                                    
# stacktrace:                                                                                                       
  - /LazyVim/lua/lazyvim/plugins/lsp/init.lua:123 _in_ **config**                                                   
  - vim/_editor.lua:0 _in_ **cmd**                                                                                  
  - /persistence.nvim/lua/persistence/init.lua:79 _in_ **load**                                                     
  - lua/:1                                                                                                                                                                                                                  
```

**Comment by glepnir** [comment]
signs only available in head mean 0.10dev.

**Comment by Innei** [comment]
Thank you!
